src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Composition;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    [Export(typeof(RequestExecutionQueue)), Shared]
+    internal partial class RequestExecutionQueue
+    {
+        private readonly AsyncQueue<QueueItem> _queue = new AsyncQueue<QueueItem>();
+        private readonly ILspSolutionProvider _solutionProvider;
+        private Solution? _lastMutatedSolution;
```

Comment:
doc threading invariants.

Reply:
I'm not sure what you mean by this. There is no concurrency here because ProcessRequestQueue is the only thing that accesses this. Or do you just mean doc that?

Reply:
Ended up moving this into the ProcessQueue method